components/Banner/examples/supplemental/MDCBannerView.h

```diff
@@ -0,0 +1,51 @@
+// Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <UIKit/UIKit.h>
+
+@protocol MDCBannerViewDelegate <NSObject>
+
+- (void)didTapButtonOnMDCBannerView:(UIButton *)button;
+
+@end
+
+@interface MDCBannerView : UIView
+
+@property(nonatomic, copy) NSString *text;
+
+@property(nonatomic, strong, nullable) UIImage *image;
+
+@property(nonatomic, readwrite, weak) id<MDCBannerViewDelegate> delegate;
+
+@property(nonatomic, strong, nullable) UIFont *buttonTitleFont;
+@property(nonatomic, strong, nullable) UIColor *buttonTitleColor;
+
+@property(nonatomic, strong, nullable) UIFont *textFont;
+@property(nonatomic, strong, nullable) UIColor *textColor;
+
++ (MDCBannerView *)bannerWithText:(NSString *)text contentSize:(CGSize)contentSize;
+
++ (MDCBannerView *)bannerWithText:(NSString *)text
+                            image:(UIImage *)image
+                          buttons:(NSArray<UIButton *> *)buttons
+                      contentSize:(CGSize)contentSize
```

Comment:
Do we need contentSize?
I would expect the content size to be dictated by the content (Image, Text, Buttons), not explicitly set.

Reply:
I removed it in a later commit. The frame size will be decided by all its subviews (including image..) and also user defined width, which will be passed in by view controller.